test/sql/test_metadata.py

```diff
@@ -4755,16 +4792,52 @@ def test_schematype_ck_name_boolean_not_on_name(self):
             schema.CreateTable(u1),
             'CREATE TABLE "user" ('
             "x BOOLEAN, "
-            "CONSTRAINT ck_user_x CHECK (x IN (0, 1))"
+            "CONSTRAINT type_ck_user_x CHECK (x IN (0, 1))"
             ")",
         )
 
-    def test_schematype_ck_name_enum(self):
+    def test_schematype_ck_name_boolean_not_on_name__no_type_ck(self):
+        """If a CheckConstraint is IMPLICITLY created for a Boolean column, it
+        should use the `type_ck_` prefix.
+
+        Ensure the correct naming_convention is chosen when `ck` is present but
+        `type_ck` is not.
+
+        This should default back to `ck`, but `ck` contains a `constraint_name`
+        token, which should raise an exception.
+        """
         m1 = MetaData(
-            naming_convention={"ck": "ck_%(table_name)s_%(constraint_name)s"}
+            naming_convention={"ck": "ck_%(table_name)s_%(constraint_name)s",
+                               }
+        )
+        u1 = Table("user", m1, Column("x", Boolean()))
+        # constraint is not hit
+        eq_(
+            [c for c in u1.constraints if isinstance(c, CheckConstraint)][
+                0
+            ].name,
+            "_unnamed_",
+        )
+        # but is hit at compile time
+        assert_raises_message(
+            exc.InvalidRequestError,
+            r"Naming convention including \%\(constraint_name\)s token "
```

Comment:
OK so, more monkey wrenches, when the user gets this error, the thing they likely *want* to do is add a "type_ck" convention to their dictionary, because now they can!   So the message should refer to that.   Since this is completely confusing, you an also add a "code" to the exception to point them to a doc blurb at https://docs.sqlalchemy.org/en/13/errors.html that explains the whole story.     this one is a real quagmire.

Reply:
will do!